napari/layers/shapes/_shape_list.py

```diff
@@ -571,6 +748,19 @@ def update_face_color(self, index, face_color, update=True):
         if update:
             self._update_displayed()
 
+    def update_face_colors(self, indices, face_colors, update=True):
+        """same as update_face_color() but for multiple indices/facecolors at once"""
+        self._face_color[indices] = face_colors
+        all_indices = np.bitwise_and(
+            np.isin(self._mesh.triangles_index[:, 0], indices),
+            self._mesh.triangles_index[:, 1] == 0,
+        )
+        self._mesh.triangles_colors[all_indices] = self._face_color[
+            self._mesh.triangles_index[all_indices, 0]
+        ]
+        if update:
+            self._update_displayed()
```

Comment:
Note for later (not this pr),
I see this pattern used in many places, I _think_ it would be better to turn it into a reentrant context manager `self.batch_update()` that execute all the `_update_display`/ `_update_z_order` on exit.